test/Schema/Coupon/index.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const assert = require('assert');
+const main = require('../../../src');
+const Coupon = main.Coupon;
+
+describe('Schema/Coupon', function () {
+    it('Coupon isLimitReached should return true when usedCount is more than the max', function () {
```

Comment:
You should test all the other alternatives - if there is no usedCountMax, or if useCount does not exceed usedCountMax what will be the result